### DISCUSSION TITLE
Discussion on the naming and structure of the MATLAB folder and scripting folders

### DISCUSSION THREAD

**Discussion by lmalenfant** [VirtualPhotonics/VTS#97] [General]
I pulled the scripting branch https://github.com/VirtualPhotonics/VTS/tree/feature/89-net-polyglot-notebook-and-jupyter-notebook-samples and noticed that the MATLAB directory structure has been modified and everything is now in an examples folder. I think this requires a bigger discussion with the whole group.

@dcuccia & @hayakawa16, I created this discussion so we can do that here.

**Comment by dcuccia** [comment]
Yep, it was the same category of things - "external" script calling the Vts library, so I made the root more general, and allowed for future growth. This was discussed in the design before I began implementing. Now that we're adding new types of script examples, I think this is a good idea, in order to avoid polluting the root. I'd also adjusted the related Powershell docs accordingly, and have made them easier to change by explicitly calling out a root directory.

**Reply by lmalenfant** [reply]
I thought the scripting samples (other than C#) were going to be pulled out of the repository once they were complete, this structure seems like they will remain here. What else do you anticipate putting in the "examples" folder? Maybe scripting should be at the root?

**Reply by lmalenfant** [reply]
The PowerShell scripts now fail to run so I think it would be a good idea to modify the folder structure and the PowerShell scripts in a separate issue and then either myself or @hayakawa16 can work on it because we have MATLAB and are able to make sure nothing is broken.

**Reply by dcuccia** [reply]
I'm actively grant writing and stuck on a pythonnet issue, so feel free to work on this branch. Just trying to group relevant concerns. Matlab scripts make sense to move out of the repo, too, for the same reasons as the others.

**Reply by lmalenfant** [reply]
@dcuccia I fixed the PowerShell scripts to work with the folder structure and after discussing with @hayakawa16 we decided to go with "scripting" at the root. Our reasoning for this is that some of the code in "Matlab" does not fall under the category of "examples" but scripting is good. I committed and pushed my changes.

**Reply by dcuccia** [reply]
Ok. My thought in creating that structure was that scripting was only one type of example of how to exercise the core VTS, but there could be apps, too. But that's fine - maybe it will be short lived in any case.

**Reply by dcuccia** [reply]
Thanks for fixing the PS, BTW.

**Reply by lmalenfant** [reply]
> Ok. My thought in creating that structure was that scripting was only one type of example of how to exercise the core VTS

If that changes, we can revisit.
